### PR TITLE
Some contract updates

### PR DESCRIPTION
**Pull request by vzotova**
* deposits  from worklock allowed only after initialization
* `setSnapshots` method for `StakingInterface`
* check snapshots in `batchDeposit`
* allow `setWindDown` without deposit

### CODE REVIEW

File: nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol [hunk 1]
```diff
@@ -540,7 +540,7 @@ contract StakingEscrow is Issuer, IERC900History {
         uint256 _value,
         uint16 _periods
     )
-        external
+        external isInitialized
```
Comment (cygnusv) [inline nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol:543]: Simple and elegant :)

File: nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol [hunk 2]
```diff
@@ -660,7 +660,7 @@ contract StakingEscrow is Issuer, IERC900History {
             uint256 endIndex = j + numberOfSubStakes;
             require(numberOfSubStakes > 0 && subStakesLength >= endIndex);
             StakerInfo storage info = stakerInfo[staker];
-            require(info.subStakes.length == 0);
+            require(info.subStakes.length == 0 && !info.flags.bitSet(SNAPSHOTS_DISABLED_INDEX));
```
Comment (cygnusv) [inline nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol:663]: What's the rationale for this change?

Comment (vzotova) [reply nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol:663]: Because we already have code that assumes that snapshots were enabled (in `batchDeposit`) and `setSnapsots` works for everybode (not only staker), I see that we need to protect one or another method

**Review by derekpierre — APPROVED** [review]
🎸

**Review by cygnusv — APPROVED** [review]

**Review by KPrasch — APPROVED** [review]
